src/Nancy.Tests.Functional/Modules/PerRouteAuthModule.cs

```diff
@@ -19,21 +17,14 @@ public PerRouteAuthModule()
 
             Get["/requiresclaims"] = _ =>
             {
-                this.RequiresClaims(new[] { "test", "test2" });
+                this.RequiresClaims(c => c.Type == "test", c => c.Type == "test2");

```

Comment:
It would have to be called 'RequiredClaimTypes'.

Regardless, its not particularly useful as you generally need to inspect
_at least_ 2 attributes of a claim - the Type and the Value. See HasClaim
https://msdn.microsoft.com/en-us/library/hh194432%28v=vs.110%29.aspx and
the predicate overload
https://msdn.microsoft.com/en-us/library/hh159713(v=vs.110).aspx
 On 26 Feb 2015 23:01, "Andreas Håkansson" <notifications@github.com> wrote:

> In src/Nancy.Tests.Functional/Modules/PerRouteAuthModule.cs
> <https://github.com/NancyFx/Nancy/pull/1820#discussion_r25469286>:
>
> > @@ -19,21 +17,14 @@ public PerRouteAuthModule()
> >
> >              Get["/requiresclaims"] = _ =>
> >              {
> > -                this.RequiresClaims(new[] { "test", "test2" });
> > +                this.RequiresClaims(c => c.Type == "test", c => c.Type == "test2");
>
> Perhaps we could have an overload of RequireClaims that takes a params
> string[] claim. The overload would iterate over the params and compare
> them against the Type of each claim. It would give us a nicer syntax.
> Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NancyFx/Nancy/pull/1820/files#r25469286>.
>